### PR TITLE
dev/core#183 Convert creating of temporary tables in CRM/Mailing file…

### PR DESCRIPTION
**Pull request by seamuslee001**
…s to using CRM_Utils_SQL_TempTable

Overview
----------------------------------------
This converts the creation of temporary tables in CRM/Mailing to using the CRM_Utils_SQL_TempTable interface

Before
----------------------------------------
Legacy methods of creating temporary tables used

After
----------------------------------------
CRM_Utils_SQL_TempTable interface used

ping @eileenmcnaughton @totten @monishdeb

### CODE REVIEW

**Comment by seamuslee001** [comment]
@eileenmcnaughton both temp table creates were covered by unit tests just FYI

**Comment by seamuslee001** [comment]
Jenkins re test this please

File: CRM/Mailing/BAO/Recipients.php [hunk 1]
```diff
@@ -115,14 +115,13 @@ public static function updateRandomRecipients($sourceMailingId, $newMailingID, $
     if ($totalLimit) {
       $limitString = "LIMIT 0, $totalLimit";
     }
-    CRM_Core_DAO::executeQuery("DROP TEMPORARY TABLE IF EXISTS  srcMailing_$sourceMailingId");
+    $temporaryTable = CRM_Utils_SQL_TempTable::build()
+      ->setCategory('srcmailing' . $sourceMailingId)
+      ->setMemory()
+      ->createWithColumns("mailing_recipient_id int unsigned, id int PRIMARY KEY AUTO_INCREMENT, INDEX(mailing_recipient_id)");
+    $temporaryTableName = $temporaryTable->getName();
```
Comment (eileenmcnaughton) [inline CRM/Mailing/BAO/Recipients.php]: should be single quotes - but OK

**Comment by eileenmcnaughton** [comment]
@seamuslee001 by both - you mean the first 2?

**Comment by seamuslee001** [comment]
@eileenmcnaughton by both i mean both files

**Comment by seamuslee001** [comment]
@eileenmcnaughton the only one not covered is the one in `CRM/Mailing/Event/BAO/Delivered.php`

**Comment by eileenmcnaughton** [comment]
Erm - but there are 3

api_v3_MailingTest covers the first pretty well

**Comment by seamuslee001** [comment]
@eileenmcnaughton the AB tests one cover the one in the Recipients file as per https://test.civicrm.org/job/CiviCRM-Core-PR/30509/

**Comment by eileenmcnaughton** [comment]
Ok - I have a test now to step through it so merging this

**Pull request merged**